src/library/previewbuttondelegate.cpp

```diff
@@ -134,64 +164,64 @@ QSize PreviewButtonDelegate::sizeHint(const QStyleOptionViewItem& option,
+        DEBUG_ASSERT(index != m_currentEditedCellIndex);
+        if (m_currentEditedCellIndex.isValid()) {
+            // Close the editor in the other cell
+            parentTableView()->closePersistentEditor(m_currentEditedCellIndex);
         }
-        m_pTableView->openPersistentEditor(index);
-        m_isOneCellInEditMode = true;
+        parentTableView()->openPersistentEditor(index);
         m_currentEditedCellIndex = index;
-    } else if (m_isOneCellInEditMode) { // close editor if the mouse leaves the button
-        m_isOneCellInEditMode = false;
-        m_pTableView->closePersistentEditor(m_currentEditedCellIndex);
+    } else if (m_currentEditedCellIndex.isValid()) {
+        // Close editor if the mouse leaves the button
+        parentTableView()->closePersistentEditor(m_currentEditedCellIndex);
         m_currentEditedCellIndex = QModelIndex();
     }
 }
 
 void PreviewButtonDelegate::buttonClicked() {
-    if (!m_pTableView) {
+    VERIFY_OR_DEBUG_ASSERT(m_currentEditedCellIndex.isValid()) {
         return;
     }
-
-    TrackModel *pTrackModel = dynamic_cast<TrackModel*>(m_pTableView->model());
-    if (!pTrackModel) {
+    TrackModel* const pTrackModel = trackModel(parentTableView());
+    VERIFY_OR_DEBUG_ASSERT(pTrackModel) {
         return;
     }
 
-    QString group = PlayerManager::groupForPreviewDeck(0);
+    QString group = PlayerManager::groupForPreviewDeck(kPreviewDeckIndex);
     TrackPointer pOldTrack = PlayerInfo::instance().getTrackInfo(group);
-    bool playing = m_pPreviewDeckPlay->toBool();
 
     TrackPointer pTrack = pTrackModel->getTrack(m_currentEditedCellIndex);
     if (pTrack && pTrack != pOldTrack) {
         emit loadTrackToPlayer(pTrack, group, true);
-    } else if (pTrack == pOldTrack && !playing) {
+    } else if (pTrack == pOldTrack && !isPreviewDeckPlaying()) {
         // Since the Preview deck might be hidden
         // Starting at cue is a predictable behavior
         m_pCueGotoAndPlay->set(1.0);
     } else {
         m_pPreviewDeckPlay->set(0.0);
-
     }
 }
 
 void PreviewButtonDelegate::previewDeckPlayChanged(double v) {
-    m_pTableView->update();
-    if (m_isOneCellInEditMode) {
-        TrackModel *pTrackModel = dynamic_cast<TrackModel*>(m_pTableView->model());
-        if (!pTrackModel) {
-            return;
-        }
-        QString group = PlayerManager::groupForPreviewDeck(0);
-        TrackPointer pPreviewTrack = PlayerInfo::instance().getTrackInfo(group);
-        TrackPointer pTrack = pTrackModel->getTrack(m_currentEditedCellIndex);
-        if (pTrack && pTrack == pPreviewTrack) {
-            emit buttonSetChecked(v > 0.0);
-        }
+    parentTableView()->update();
+    if (!m_currentEditedCellIndex.isValid()) {
+        return;
+    }
+    TrackModel* const pTrackModel = trackModel(parentTableView());
+    VERIFY_OR_DEBUG_ASSERT(pTrackModel) {
+        return;
+    }
+    QString group = PlayerManager::groupForPreviewDeck(kPreviewDeckIndex);
```

Comment:
The whole group can become a constant implicite shared between all theses delegates.